puppet-mode.el

```diff
@@ -225,6 +243,48 @@ Return nil, if there is no special context at POS, or one of
   "Determine whether POS is inside a string or comment."
   (not (null (puppet-syntax-context pos))))
 
+(defun puppet-get-repl-proc ()
+  (unless (comint-check-proc puppet-repl-buffer)
+    (puppet-repl)
+    ;; see issue https://github.com/defunkt/coffee-mode/issues/332
```

Comment:
Not sure if this REPL will have the same issues as the coffee one. I don't recall such hacks in `inf-clojure` for instance.